src/main/java/appeng/tile/inventory/AppEngInternalAEInventory.java

```diff
@@ -150,42 +141,85 @@ public ItemStack getStackInSlot( final int var1 )
 	}
 
 	@Override
-	public ItemStack decrStackSize( final int slot, final int qty )
+	public ItemStack insertItem( int slot, ItemStack stack, boolean simulate ) 
+	{
+		 if ( stack.isEmpty() )
+		 {
+	            return ItemStack.EMPTY;
+		 }
+
+		 ItemStack existing = getStackInSlot( slot );
+		 int limit = getStackLimit( slot, stack );
+
+		 if ( !existing.isEmpty() )
+	     {
+			 if ( !ItemHandlerHelper.canItemStacksStack( stack, existing ) )
+				 return stack;
+
+			 limit -= existing.getCount();
+	     }
+
+		 if ( limit <= 0 )
+			 return stack;
+
+		 boolean reachedLimit = stack.getCount() > limit;
+
+	     if ( !simulate )
+	     {
+	    	 if ( existing.isEmpty() )
+	         {
+	    		 this.inv[slot] = AEApi.instance().storage().createItemStack( reachedLimit ? ItemHandlerHelper.copyStackWithSize( stack, limit ) : stack );
+	         }
+	         else
+	         {
+	        	 existing.grow( reachedLimit ? limit : stack.getCount() );
+	         }
+	    	 if( this.te != null && Platform.isServer() )
+	    	 {
+				this.te.onChangeInventory( this, slot, InvOperation.insert, ItemStack.EMPTY,  reachedLimit ? ItemHandlerHelper.copyStackWithSize( stack, limit) : stack );
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') ')' is not preceded with whitespace. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=checkstyle%3Acom.puppycrawl.tools.checkstyle.checks.whitespace.ParenPadCheck)